### PR TITLE
Add comments to the Repository/Timeline traits.

### PR DESCRIPTION
**Pull request by hlinnaka**
Let's try to have comments on every public function. This doesn't quite
get us there yet, but close.

### CODE REVIEW

File: pageserver/src/repository.rs [hunk 1]
```diff
@@ -33,12 +45,30 @@ pub trait Timeline {
     /// Does relation exist?
     fn get_relsize_exists(&self, tag: RelTag, lsn: Lsn) -> Result<bool>;
 
-    // Functions used by WAL receiver
+    //------------------------------------------------------------------------------
+    // Public PUT functions, to update the repository with new page versions.
+    //
+    // These are called by the WAL receiver to digest WAL records.
+    //------------------------------------------------------------------------------
```
Comment (hlinnaka) [inline pageserver/src/repository.rs:52]: @ericseppanen, is there some Rust convention for "section divider" comments like this?

**Comment by hlinnaka** [comment]
@arssher, since you're more new to the code, what do you think would be helpful to get an overview? Or some important detail thtat's missing?

**Review by knizhnik — COMMENTED** [review]
I still concerning about resurrection of multi-tenancy support.
You have renamed System to Timeline and now Timeline seems to be the main storage unit. But timeline is by definition a member object: it is part of some more container (timeline of what?).  

So if we will return notion of System/User/Instance/..., then how it will affect the existed hierarchy of classes. Will there be some top level class i.e. Instance with several underlying Timelines? Or we do not need such hierarchy? In this case I think we need some other notion rather than Timeline.

File: pageserver/src/repository.rs [hunk 2]
```diff
@@ -11,9 +11,17 @@ use zenith_utils::lsn::Lsn;
 /// A repository corresponds to one .zenith directory. One repository holds multiple
 /// timelines, forked off from the same initial call to 'initdb'.
```
Comment (arssher) [inline pageserver/src/repository.rs:12]: The fork point is not necessary the same initial initdb, we can branch at any moment/timeline, right? Well, I guess this says initdb is the deepest common ancestor of everyone...

**Review by arssher — APPROVED** [review]
Looks good to me. I wondered shouldn't e.g. guts of `run_branch_cmd` (`create_branch`, `find_end_of_wal`) be methods of `Repository`), but `Repository` seems to have narrower meaning for now.

File: pageserver/src/repository.rs (hunk 2 above)
Comment (hlinnaka) [reply pageserver/src/repository.rs:12]: > Well, I guess this says initdb is the deepest common ancestor of everyone...

Right. Although we've also chatted about allowing importing completely unrelated postgres clusters to the same page server or repository. Nothing in the page server depends on all timelines sharing a common ancestor, there just isn't any UI to do that currently.

**Comment by hlinnaka** [comment]
> I still concerning about resurrection of multi-tenancy support.
> You have renamed System to Timeline and now Timeline seems to be the main storage unit. But timeline is by definition a member object: it is part of some more container (timeline of what?).
> 
> So if we will return notion of System/User/Instance/..., then how it will affect the existed hierarchy of classes. Will there be some top level class i.e. Instance with several underlying Timelines? Or we do not need such hierarchy? In this case I think we need some other notion rather than Timeline.

Timelines live in a Repository now. Currently, a page server can serve exactly one Repository, see `page_cache.rs`, but we might implement multi-tenancy by allowing one page server to serve multiple repositories.

Of course, plans might change :-). Let's keep it simple for now and cross the bridge when we get there.

File: pageserver/src/repository.rs (hunk 1 above)
Comment (ericseppanen) [reply pageserver/src/repository.rs:52]: Not that I know of.

**Review by ericseppanen — APPROVED** [review]
These are really great, thanks!

**Pull request merged**